src/components/leaf.js

```diff
@@ -148,17 +145,41 @@ class Leaf extends React.Component {
     const hasFocus = selection.hasFocusBetween(node, start, end)
     if (!hasAnchor && !hasFocus) return
 
+    // We have a selection to render, so prepare a few things...
+    const ref = ReactDOM.findDOMNode(this)
+    const el = findDeepestNode(ref)
+    const window = getWindow(el)
+
+    // If no selection is defined we're probably in a node environment and we can skip the selection magic.
+    if (!window || !window.getSelection) return
+
+    const native = window.getSelection()
+    const parent = ref.closest('[contenteditable]')
+
+    // If the selection is blurred but the native selection has still edges in this leaf,
+    // we need to clean up the ranges and blur the contenteditable.
+    if (selection.isBlurred && (hasAnchor || hasFocus)) {
```

Comment:
The problem is that I thought that `hasAnchor || hasFocus` is being computed by the native selection, but it's computed using the document selection. this obviously yields true multiple times if multiple slate instances exist and then causes the removeallranges thing to remove the selection of the second slate instance.

The idea behind this was that by checking if the current selection is still in the contenteditable although it should be blurred, we blur the selection using the DOM.